azure-pipelines.yml

```diff
@@ -23,3 +23,72 @@ jobs:
   parameters:
     name: WindowsPy27
     vmImage: vs2017-win2016
+
+- job: 'Checks_and_doc'
+  pool:
+    vmImage: ubuntu-16.04
+  steps:
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: '3.7'
+      architecture: 'x64'
+
+  - script: |
+      sudo apt-get install -y libc6-dev-i386
+      ci/incremental/install_miniconda.sh
+      export PATH=$HOME/miniconda3/bin:$PATH
+      export CONDA_ENV=pandas
+      export ENV_FILE=ci/deps/azure-checks-and-doc.yaml
+      ci/incremental/setup_conda_environment.sh
+    displayName: 'Set up environment'
+
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      export CONDA_ENV=pandas
+      ci/incremental/build.sh
+    displayName: 'Build'
+    condition: true
+
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      export AZURE=true
+      ci/code_checks.sh lint
+    displayName: 'Linting'
+    condition: true
+
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      export AZURE=true
+      ci/code_checks.sh patterns
+    displayName: 'Looking for unwanted patterns'
+    condition: true
+
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      ci/code_checks.sh doctests
+    displayName: 'Running doctests'
+    condition: true
+
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      source activate pandas
+      doc/make.py html
+    displayName: 'Building docs'
+    condition: true
+
+  - script: |
+      export REPO_DIR="/tmp/pandas-docs-travis"
+      git config --global user.email "pandas-docs-bot@localhost"
+      git config --global user.name "pandas-docs-bot"
+      # TODO PANDAS_GH_TOKEN needs to be added as a secret variable
```

Comment:
I think @jorisvandenbossche can generate a key for the pandas docs bot.

Reply:
Good to know. I'll probably do some tests with a repo of my own first. I still need to find how to add it to azure, besides generating it.

Besides moving the doc building to azure, the main feedback I'd like is about publishing the doc build in this way.

@jorisvandenbossche mentioned about scikit-learn, which published in CircliCI infrastructure. But not sure if adding directories to a github pages repo is simpler.